### PR TITLE
Improve security of storage account

### PR DESCRIPTION
**Pull request by johndowns**
This PR improves the security of the temporary storage account in the following ways:

- Switch to Entra authentication for all interaction with the storage account.
- Block use of the storage account keys and require Entra authentication.
- Deny public network access to the storage account, and grant access to the temporary Application Gateway VNet as well as the deployer's public IP address.
- Wait for the Application Gateway to be ready before proceeding with the certificate generation.
- Remove the example secrets in the README.md file and replace with permitted example secrets.

These changes also mean that the sample can now be run in Microsoft's own subscriptions.

### CODE REVIEW

File: generate-cert.sh [hunk 1]
```diff
@@ -25,7 +30,10 @@ az storage blob service-properties update --account-name $STORAGE_ACCOUNT_NAME -
 
 echo "Uploading placeholder to storage"
 echo pong>ping.txt
-az storage blob upload --account-name $STORAGE_ACCOUNT_NAME -c \$web -n ping -f ./ping.txt --auth-mode key
+az storage blob upload --account-name $STORAGE_ACCOUNT_NAME -c \$web -n ping -f ./ping.txt --auth-mode login
+
+echo "Waiting for website to be ready"
+curl --retry 10 --retry-delay 10 -s -4 http://$FQDN/ping
```
Comment (ferantivero) [inline generate-cert.sh:36]: great addition @johndowns

File: README.md [hunk 2]
```diff
@@ -89,12 +89,25 @@ It is based on [Let's Encrypt®](https://letsencrypt.org). Let's Encrypt is a no
...
+   ```output
+   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+
+   Create a file containing just this data:
+
+   A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u.C2dE3fH4iJ5kL6mN7oP8qR9sT0uV1w
+
+   And make it available on your web server at this URL:
+
+   http://mysubdomain.eastus.cloudapp.azure.com/-well-known/acme-challenge/A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u
```
[the remark this answers is not in the document]
Comment (johndowns) [reply README.md:101]: Are you suggesting changing it just in the readme, or in the actual scripts?

File: README.md [hunk 3]
```diff
@@ -89,12 +89,25 @@ It is based on [Let's Encrypt®](https://letsencrypt.org). Let's Encrypt is a no
...
+   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+
+   Press Enter to Continue
+   ```
 
    1. Create a file name with the name presented by Certbot during the execution, with a `.txt` extension and add the content inside as instructed. For example,
 
    ```bash
-   echo "-Nahn2wS1fLeqGwqjDBIWxSpL5U4mlb_oA50wsPeoqk.T_a4tluV9By4PqiMY4Xz5iLe5ty1whK_vNK21LY6ZTU">-Nahn2wS1fLeqGwqjDBIWxSpL5U4mlb_oA50wsPeoqk.txt
+   echo "A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u.C2dE3fH4iJ5kL6mN7oP8qR9sT0uV1w">A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u.txt
```
[the remark this answers is not in the document]
Comment (johndowns) [reply README.md:110]: This is related to the SFI work going on. We need to remove any "real" secrets - even if they're expired or no longer valid - and can only use specific allowed secret values.

File: authenticator.sh [hunk 4]
```diff
@@ -6,6 +6,6 @@ echo $CERTBOT_VALIDATION>${CERTBOT_TOKEN}.txt
 
 echo "Uploading validation token to $STORAGE_ACCOUNT_NAME"
 
-az storage blob upload --account-name $STORAGE_ACCOUNT_NAME -c \$web -n "${CERTBOT_TOKEN}" -f "./${CERTBOT_TOKEN}.txt" --auth-mode key --only-show-errors
+az storage blob upload --account-name $STORAGE_ACCOUNT_NAME -c \$web -n "${CERTBOT_TOKEN}" -f "./${CERTBOT_TOKEN}.txt" --auth-mode login --only-show-errors
```
[the remark this answers is not in the document]
Comment (johndowns) [reply authenticator.sh:9]: If you're already signed into the CLI (which you would need to be to run the deploy steps in the _generate-cert.sh_ file), then this step inherits the auth context and you won't be prompted again. Is that what you mean?

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ferantivero) [reply README.md:101]: definitely in the README where I personally think it is most important part to understand this conceptually.

What Im not quite sure is whether we want to do this change. I mean, it was just a suggestion.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ferantivero) [reply README.md:110]: provided this is a secret, should this be redacted? or what we updated in here is a fake value?

File: authenticator.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (ferantivero) [reply authenticator.sh:9]: awesome we are ensuring as part of the steps is not going to be prompted.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (johndowns) [reply README.md:110]: This is a fake value, from the list of allowable fake secret values.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (johndowns) [reply README.md]: Yeah I can see the benefit. Updated.

**Review by ferantivero — APPROVED** [review]
great additions and improvements in here @johndowns, LGTM 🚢

**Comment by ferantivero** [comment]
#sign-off please let's squash and merge when ready @johndowns

**Pull request merged**